crates/sui-core/src/authority.rs

```diff
@@ -2393,6 +2394,15 @@ impl AuthorityState {
                     .await?
             }
         };
+        // skip one event if exclusive cursor is provided,
+        // otherwise truncate to the original limit.
+        if cursor.is_some() {
+            if !stored_events.is_empty() {
+                stored_events.remove(0);
+            }
+        } else {
+            stored_events.truncate(limit - 1);
+        }
```

Comment:
Why do we do this only for `events` but not for other indexers in authority.rs ? This would be very confusing to the caller of the authority.rs.

It might be worth considering changing the return types for methods in sui-storage/src/indexes to include `has_next_page` and `next_cursor` to have better encapsulation of this pagination logic, so that it's absolutely clear to the callers(both in authority.rs and rpcs). What's your thoughts about this? cc @patrickkuo

Reply:
> methods in sui-storage/src/indexes to include has_next_page and next_cursor

Pagination imo is a pure JSON RPC logic, sui-storage should not care about it, the changes of PR in sui-storage is only changing cursor from inclusive to exclusive.

>  only for events but not for other pagination api in authority.rs ?

the other alternative is to shift starting index in each SQLite function of `events_by_transaction`, `events_by_module_id` ... like this PR does for functions like `get_transactions_*`, as we will remove SQLite soon, this is just quicker.

Reply:
> sui-storage should not care about it, the changes of PR in sui-storage is only changing cursor from inclusive to exclusive.

We are effectively using pagination in sui-storage with the cursor and limit in the API. My goal is to make sure there's no ambiguity in the definition of cursor and what the API returns for functions defined in `authority.rs` and `sui-torage` from the perspective of callers. Otherwise, it would be easy to introduce accidental bugs during refactoring or adding new endpoints. 

It also works if you just add detailed comments about the behavior in front of each of the public functions related to pagination. 

> as we will remove SQLite soon, this is just quicker.

Sounds good given that we will remove this soon. Can you still add some docs comments?